components/dashboard/src/teams/TeamSettings.tsx

```diff
@@ -397,3 +406,78 @@ function OrgDefaultWorkspaceImageModal(props: OrgDefaultWorkspaceImageModalProps
         </Modal>
     );
 }
+
+interface WorkspaceClassOptionsProps {
+    settings: OrganizationSettings | undefined;
+    disabled: boolean;
+}
+const WorkspaceClassOptions = (props: WorkspaceClassOptionsProps) => {
+    const [errorMsg, setErrorMsg] = useState("");
+    const [selectedValue, setSelectedValue] = useState(props.settings?.allowedWorkspaceClasses ?? []);
+    const updateTeamSettings = useUpdateOrgSettingsMutation();
+    const { data: classes, isError, isLoading } = useWorkspaceClasses();
+
+    const { toast } = useToast();
+    const handleUpdateTeamSettings = useCallback(
+        async (classes: string[]) => {
+            try {
+                await updateTeamSettings.mutateAsync(
+                    {
+                        ...props.settings,
+                        allowedWorkspaceClasses: classes,
+                    },
+                    {
+                        onSuccess: () => {
+                            toast({ message: "Available workspace classes updated for your organization." });
+                        },
+                    },
+                );
+            } catch (error) {
+                if (!ErrorCode.isUserError(error["code"])) {
+                    console.error(error);
+                }
+                setErrorMsg(error.message);
+            }
+        },
+        [updateTeamSettings, props, toast],
+    );
+
+    if (isError || !classes) {
+        return <div>Something went wrong</div>;
+    }
+
+    if (isLoading) {
+        return <LoadingState />;
+    }
+
+    return (
+        <>
+            {errorMsg.length > 0 && (
+                <Alert type="error" className="mb-2">
+                    {errorMsg}
+                </Alert>
+            )}
+            <Label>
+                {classes.map((wsClass) => (
+                    <CheckboxInputField
+                        id={wsClass.id}
+                        label={wsClass.displayName}
+                        hint={wsClass.description}
+                        checked={selectedValue.includes(wsClass.id)}
```

Comment:
I revert this changes, since I'm not sure what state we need to show when user unchecked all, 
- Should it change to check all?  -> It's strange because I know that I want to uncheck it
- Or keep unchecked all -> then why first view is all checked? Will it leads to misunderstanding that allow nothing?

Update description above to make them know that select nothing means allow all kind of workspace classes

Reply:
@mustard-mh I think we should make it more explicit that you (probably) can't disable all classes. It feels like the pattern of "nothing is selected" = "everything is selected" is a bit flawed. I can see us starting off with everything checked (`.length === 0`) and then require that at least one checkbox remains checked at all times to avoid confusion.

Reply:
@filiptronicek  Update to show message that there should have at least one class slected

<img width="818" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/0d7384ee-5a19-41ef-84b8-396b9cb85dea">